app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -7,12 +7,43 @@ def make_offer
         vendor_api_token_id: @current_vendor_api_token.id,
       )
 
-      decision = MakeAnOffer.new(
-        actor: api_user,
-        application_choice: application_choice,
-        offer_conditions: params.dig(:data, :conditions),
-        course_data: params.dig(:data, :course),
-      )
+      course_data = params.dig(:data, :course)
+
+      if course_data.present?
+        course_option = GetCourseOptionFromCodes.new(
+          provider_code: course_data[:provider_code],
+          course_code: course_data[:course_code],
+          recruitment_cycle_year: course_data[:recruitment_cycle_year],
+          study_mode: course_data[:study_mode],
+          site_code: course_data[:site_code],
+        ).call
+
+        change_offer_form = ProviderInterface::ChangeOfferForm.new step: :update,
+                                                                   application_choice: application_choice,
+                                                                   provider_id: (course_option.course.provider.id if course_option),
+                                                                   course_id: (course_option.course.id if course_option),
+                                                                   course_option_id: (course_option.id if course_option)
+
+        (render_cannot_find_course_option and return) unless change_offer_form.valid?
+      else
+        course_option = nil
+      end
+
+      decision = if application_choice.offer?
+                   ChangeOffer.new(
```

Comment:
#### Coding style suggestion

`ChangeOffer` and `MakeAnOffer` constructors accept the same params so this could be reduced to:

```ruby
operation = application_choice.offer? ? ChangeOffer : MakeAnOffer
operation.new(
  actor: api_user,
  application_choice: application_choice,
  course_option_id: course_option&.id,
  offer_conditions: params.dig(:data, :conditions),
)
```